modules/prebid/ortb2blocking/module_test.go

```diff
@@ -0,0 +1,1044 @@
+								},
+							},
+						},
+					},
+				},
+				DebugMessages: []string{"Bid 1 from bidder appnexus has been rejected, failed checks: [battr]"},
+			},
+		},
+		{
+			description: "Bid blocked by multiple attribute check. Payload updated. Analytic tags successfully collected",
+			payload: hookstage.RawBidderResponsePayload{Bidder: bidder, Bids: []*adapters.TypedBid{
+				{
+					Bid: &openrtb2.Bid{
+						ID:      "1",
+						ADomain: []string{"forbidden_domain"},
+						Cat:     []string{"fishing"},
+						CatTax:  1,
+						Bundle:  "forbidden_bundle",
+						Attr:    []adcom1.CreativeAttribute{1},
+						ImpID:   impID1,
+					},
+				},
+				{
+					Bid: &openrtb2.Bid{
+						ID:      "2",
+						ADomain: []string{"allowed_domain"},
+						Cat:     []string{"moto"},
+						CatTax:  2,
+						Bundle:  "allowed_bundle",
+						Attr:    []adcom1.CreativeAttribute{2},
+						ImpID:   impID2,
+					},
+				},
+			}},
+			config: json.RawMessage(`{"attributes":{"badv":{"enforce_blocks": true}, "bcat":{"enforce_blocks": true}, "cattax":{"enforce_blocks": true}, "bapp":{"enforce_blocks": true}, "battr":{"enforce_blocks": true}}}`),
+			expectedBids: []*adapters.TypedBid{
+				{
+					Bid: &openrtb2.Bid{
+						ID:      "2",
+						ADomain: []string{"allowed_domain"},
+						Cat:     []string{"moto"},
+						CatTax:  2,
+						Bundle:  "allowed_bundle",
+						Attr:    []adcom1.CreativeAttribute{2},
+						ImpID:   impID2,
+					},
+				},
+			},
+			moduleCtx: map[string]interface{}{bidder: blockingAttributes{
+				badv:   []string{"forbidden_domain"},
+				bcat:   []string{"fishing"},
+				cattax: 2,
+				bapp:   []string{"forbidden_bundle"},
+				battr:  map[string][]int{impID1: {1}}},
+			},
+			expectedHookResult: hookstage.HookResult[hookstage.RawBidderResponsePayload]{
+				AnalyticsTags: hookanalytics.Analytics{
+					Activities: []hookanalytics.Activity{
+						{
+							Name:   enforceBlockingTag,
+							Status: hookanalytics.ActivityStatusSuccess,
+							Results: []hookanalytics.Result{
+								{
+									Status: hookanalytics.ResultStatusBlock,
+									Values: map[string]interface{}{
+										attributesAnalyticKey: []string{"badv", "bcat", "cattax", "bapp", "battr"},
+										badvAnalyticKey:       []string{"forbidden_domain"},
+										cattaxAnalyticKey:     []adcom1.CategoryTaxonomy{1},
+										bappAnalyticKey:       []string{"forbidden_bundle"},
+										battrAnalyticKey:      []int{1},
+									},
+									AppliedTo: hookanalytics.AppliedTo{Bidders: []string{bidder}, ImpIds: []string{impID1}},
+								},
+								{
+									Status:    hookanalytics.ResultStatusAllow,
+									AppliedTo: hookanalytics.AppliedTo{Bidders: []string{bidder}, ImpIds: []string{impID2}},
+								},
+							},
+						},
+					},
+				},
+				DebugMessages: []string{"Bid 1 from bidder appnexus has been rejected, failed checks: [badv bcat cattax bapp battr]"},
```

Comment:
This test case fails unexpectedly from time to time due to this string because the order of listed attributes (`[badv bcat cattax bapp battr]`) is changing. Which is caused by the data structure used under the hood, look here: https://github.com/pi-prebid/prebid-server/blob/8d16bd846035a707d940fed1409a06e2fbbb87c1/modules/prebid/ortb2blocking/hook_raw_bidder_response.go#L38
The order of the elements inside the `map` is not guaranteed to be the same. I fixed similar issue in my PR, so you can check it out: https://github.com/pi-prebid/prebid-server/blob/5f84ba85f1e39e581a43a7fa52753e1d7ab53b1e/modules/prebid/ortb2blocking/hook_bidderrequest.go#L318-L336

Reply:
thanks, I'll update the PR